lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -691,18 +676,17 @@ func getPotentialHints(cfg *SelectHopHintsCfg) ([]*channeldb.OpenChannel,
 		return nil, err
 	}
 
-	privateChannels := make([]*channeldb.OpenChannel, 0, len(openChannels))
+	privateChannels := make([]*HopHintInfo, 0, len(openChannels))
 	for _, oc := range openChannels {
-		isPublic := oc.ChannelFlags&lnwire.FFAnnounceChannel != 0
-		if !isPublic {
+		if !oc.IsPublic {
```

Comment:
nit: now that we already have the `HopHintInfo` here maybe we can do `if !oc.IsPublic && oc.IsActive {`